x-pack/plugins/security_solution/public/management/components/endpoint_responder/isolate_action.tsx

```diff
@@ -5,89 +5,47 @@
  * 2.0.
  */
 
-import React, { memo, useEffect } from 'react';
-import type { ActionDetails } from '../../../../common/endpoint/types';
-import { useGetActionDetails } from '../../hooks/endpoint/use_get_action_details';
-import type { EndpointCommandDefinitionMeta } from './types';
+import React, { memo } from 'react';
+import type { ActionRequestComponentProps } from './types';
 import { useSendIsolateEndpointRequest } from '../../hooks/endpoint/use_send_isolate_endpoint_request';
-import type { CommandExecutionComponentProps } from '../console/types';
 import { ActionError } from './action_error';
-
-export const IsolateActionResult = memo<
-  CommandExecutionComponentProps<
-    { comment?: string },
-    {
-      actionId?: string;
-      actionRequestSent?: boolean;
-      completedActionDetails?: ActionDetails;
-    },
-    EndpointCommandDefinitionMeta
-  >
->(({ command, setStore, store, status, setStatus, ResultComponent }) => {
-  const endpointId = command.commandDefinition?.meta?.endpointId;
-  const { actionId, completedActionDetails } = store;
-  const isPending = status === 'pending';
-  const actionRequestSent = Boolean(store.actionRequestSent);
-
-  const isolateHostApi = useSendIsolateEndpointRequest();
-
-  const { data: actionDetails } = useGetActionDetails(actionId ?? '-', {
-    enabled: Boolean(actionId) && isPending,
-    refetchInterval: isPending ? 3000 : false,
-  });
-
-  // Send Isolate request if not yet done
-  useEffect(() => {
-    if (!actionRequestSent && endpointId) {
-      isolateHostApi.mutate({
-        endpoint_ids: [endpointId],
-        comment: command.args.args?.comment?.[0],
-      });
-
-      setStore((prevState) => {
-        return { ...prevState, actionRequestSent: true };
-      });
+import { useUpdateActionState } from './hooks';
+
+export const IsolateActionResult = memo<ActionRequestComponentProps>(
+  ({ command, setStore, store, status, setStatus, ResultComponent }) => {
+    const endpointId = command.commandDefinition?.meta?.endpointId;
+    const { completedActionDetails, actionRequest } = store;
+    const isPending = status === 'pending';
+    const isolateHostApi = useSendIsolateEndpointRequest();
+
+    useUpdateActionState({
+      actionRequestApi: isolateHostApi,
+      actionRequest,
+      command,
```

Comment:
My guess is that when we improve this hook to support any command, that we'll instead require that the action request api `body` be provided here to the hook instead of trying to have the hook "known" about all possible command actions and building the `body` itself.

Reply:
Yeah, that would be ideal and easier to maintain.

Reply:
we agree to push this off to another PR